### PR TITLE
pagination added

### PR DESCRIPTION
**Pull request by groyvstreet**

### CODE REVIEW

File: Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/OrderRepository.cs [hunk 1]
```diff
@@ -13,9 +13,21 @@ public OrderRepository(DataContext context)
...
+
+            if (pageNumber < 1)
+            {
+                return orders;
+            }
+
+            if (pageSize < 1)
+            {
+                pageSize = 10;
+            }
```
Comment (elv1sabene) [inline Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/OrderRepository.cs]: I see that this code is repeating over and over again
May be there is a solution to write it once and use it everywhere needed

Comment (groyvstreet) [reply Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/OrderRepository.cs]: I created a generic abstract class with the protected method to which i moved this code. All repositories are inherited from this class.

File: Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/OrderRepository.cs [hunk 2]
```diff
@@ -13,9 +13,11 @@ public OrderRepository(DataContext context)
             this.context = context;
         }
 
-        public async Task<List<Order>> GetAllOrders()
+        public async Task<List<Order>> GetAllOrders(int pageNumber, int pageSize)
         {
-            return await context.Orders.AsNoTracking().ToListAsync();
+            var orders = await context.Orders.AsNoTracking().ToListAsync();
```
Comment (elv1sabene) [inline Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/OrderRepository.cs]: Insert `ToPagedList()` into chain of call, so entity framework can do pagination directly in database

File: Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/PagedRepository.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿namespace ModsenOnlineStore.Store.Infrastructure.Data
+{
+    public abstract class PagedRepository<T>
+    {
+        protected List<T> ToPagedList(List<T> entities, int pageNumber, int pageSize)
```
Comment (elv1sabene) [inline Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/PagedRepository.cs]: use `IEnumerable<T>`

File: Services/TokenAuthentication/ModsenOnlineStore.Login.Infrastructure/Data/UserRepository.cs [hunk 4]
```diff
@@ -16,8 +16,22 @@ public UserRepository(DataContext context)
...
+
+            if (pageNumber < 1)
+            {
+                return users;
+            }
+
+            if (pageSize < 1)
+            {
+                pageSize = 10;
+            }
```
Comment (elv1sabene) [inline Services/TokenAuthentication/ModsenOnlineStore.Login.Infrastructure/Data/UserRepository.cs]: pagination here?

File: Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/OrderRepository.cs (hunk 2 above)
Comment (groyvstreet) [reply Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/OrderRepository.cs]: Is it mean that i need to make ToPagedList() an extension method?

Comment (elv1sabene) [reply Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/OrderRepository.cs]: yes

**Review by elv1sabene — APPROVED** [review]

**Pull request merged**